sdk/communication/azure-communication-rooms/azure/communication/rooms/_models/_models.py

```diff
@@ -0,0 +1,224 @@
+    :vartype communication_identifier:
+    ~azure.communication.rooms._shared.models.CommunicationIdentifier
+    :ivar role: Role Name.
+    :vartype role: Optional[Union[str, RoleType]
+    """
+
+    _validation = {
+        "communication_identifier": {"required": True},
+    }
+
+    _attribute_map = {
+        "communication_identifier": {"key": "communicationIdentifier", "type": "CommunicationIdentifier"},
+        "role": {"key": "role", "type": "Optional[Union[str, RoleType]"},
+    }
+
+    def __init__(
+        self,
+        *,
+        communication_identifier: CommunicationIdentifier,
+        role: Optional[Union[str, RoleType]]=None,
+        **kwargs
+    ):
+        """
+        :param communication_identifier: Identifies a participant in Azure Communication services. A
+         participant is, for example, an Azure communication user. This model must be interpreted as a
+         union: Apart from rawId, at most one further property may be set. Required.
+        :type communication_identifier:
+         ~azure.communication.rooms._shared.models.CommunicationIdentifier
+        :param role: The Role of a room participant. Known values are: "Presenter", "Attendee", and
+         "Consumer".
+        :type role: Optional[Union[str, RoleType]]
+        """
+        super().__init__(**kwargs)
+        self.communication_identifier = communication_identifier
+        self.role = role
+
+    @classmethod
+    def from_room_participant_internal(cls, room_participant_internal: RoomParticipantInternal, **kwargs):
+        return cls(
+            communication_identifier=_CommunicationIdentifierConverter
+                .to_communication_identifier(room_participant_internal.communication_identifier),
+            role=room_participant_internal.role,
+            **kwargs
+        )
+
+    def to_room_participant_internal(self):
+        return RoomParticipantInternal(
+            communication_identifier=_CommunicationIdentifierConverter
+                .to_communication_identifier_model(self.communication_identifier),
+            role=self.role
+        )
+
+class _CommunicationIdentifierConverter():
+    @staticmethod
+    def to_communication_identifier_model(communication_identifier: CommunicationIdentifier):
+        if communication_identifier.kind == CommunicationIdentifierKind.COMMUNICATION_USER:
+            return CommunicationIdentifierModel(
+                communication_user=CommunicationUserIdentifierModel(
+                    id=communication_identifier.properties['id']
+                )
+            )
+        if communication_identifier.kind == CommunicationIdentifierKind.UNKNOWN:
+            return CommunicationIdentifierModel(
+                raw_id=communication_identifier.raw_id
+            )
+
+        raise TypeError('The type of communication identifier "{}" is not supported'
+            .format(communication_identifier.kind))
```

Comment:
Where is this method called?
Is this always called on request? On response? Or both?

Reply:
This convertor is used to convert between client-facing `Common.CommunicationIdentifier` and the `CommunicationIdentifierModel` rooms service uses on the server side. They are used every time a client passes in or receives `Participant` related info. eg. `UpdateRoom`, `AddParticipants`